### PR TITLE
Mapping policy tests

### PR DESCRIPTION
**Pull request by imLew**
**Proposed changes**:
- Added tests for the MappingPolicy. The tests cover the cases in which the policy will predict an action and the case in which it does not predict because the `policy` attribute of the last action does not match MappingPolicy.
Also added a debug message to be printed in that case.

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: tests/core/test_policies.py [hunk 1]
```diff
@@ -205,6 +220,60 @@ def create_policy(self, featurizer, priority):
...
+            self._get_next_action(policy, events, domain_with_mapping)
+            == intent_mapping[1]
+        )
+
+    def test_predict_action_listen(self, priority, domain_with_mapping, intent_mapping):
+        policy = self.create_policy(None, priority)
+        events = [
+            ActionExecuted(ACTION_LISTEN_NAME),
+            user_uttered(intent_mapping[0], 1),
+            ActionExecuted(intent_mapping[1], policy="MappingPolicy"),
```
Comment (erohmensing) [inline tests/core/test_policies.py]: Not sure the `policy="MappingPolicy"` is doing what it should in this test -- I can change it to anything and the test still passes

File: tests/core/test_policies.py [hunk 2]
```diff
@@ -205,6 +220,60 @@ def create_policy(self, featurizer, priority):
...
+    ):
+        policy = self.create_policy(None, priority)
+        events = [
+            ActionExecuted(ACTION_LISTEN_NAME),
+            user_uttered(intent_mapping[0], 1),
+            ActionExecuted(intent_mapping[1], policy="other_policy"),
+        ]
+        tracker = get_tracker(events)
+        scores = policy.predict_action_probabilities(tracker, domain_with_mapping)
+        assert scores == [0] * domain_with_mapping.num_actions
```
Comment (erohmensing) [inline tests/core/test_policies.py]: here we test whether the policy ensemble chooses mapping policy or not, but above we only test whether the mapping policy predicts action listen. Can we in the first one actually tests if mapping policy gets picked in the ensemble?

File: tests/core/test_policies.py (hunk 1 above)
Comment (imLew) [reply tests/core/test_policies.py]: Hmm, I think this is due to how `_get_next_action` works. If you change the policy name the prediction returned by the policy should be all zeros, whereas if it is "MappingPolicy" it should return a unit vector that is 1 at the index of `action_listen`.

File: tests/core/test_policies.py (hunk 2 above)
Comment (imLew) [reply tests/core/test_policies.py]: I don't think this tests whether the policy is picked but only what it does if it is picked after an action was predicted by a different policy.
I'm not sure if we can check whether it does get picked. Most likely not in the policy though, that should be a test for the ensemble.

**Comment by imLew** [comment]
I made some changes, also to the mapping policy because default mapping (e.g. action_back) were not handled in a manner consistent with other actions.
Also you are right about the second bug not being tested, but I think this is addressed by simply changing the string to something more realistic.

**Pull request merged**